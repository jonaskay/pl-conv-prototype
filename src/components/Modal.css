.Modal {
  align-items: center;
  background: rgba(0,0,0,.288);
  bottom: 0;
  display: flex;
  left: 0;
  padding: 1rem;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 100;
}

.Modal-dialog {
  background: #424242;
  border-radius: .5rem;
  box-shadow: 
    0 11px 15px -7px rgba(0,0,0,.2), 
    0 24px 38px 3px rgba(0,0,0,.14), 
    0 9px 46px 8px rgba(0,0,0,.12);
  box-sizing: border-box;
  margin: auto;
  max-width: 20rem;
  padding: 1rem;
  position: relative;
  width: 100%;
}
