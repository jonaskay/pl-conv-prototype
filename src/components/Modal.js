import React from 'react';
import './Modal.css';

class Modal extends React.Component {
  render() {
    return (
      <div className="Modal">
        <div className="Modal-dialog">{this.props.children}</div>
      </div>
    );
  }
}

export default Modal;
