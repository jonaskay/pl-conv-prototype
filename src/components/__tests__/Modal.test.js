import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Modal from '../Modal';

it('renders a modal', () => {
  const component = ReactTestRenderer.create(<Modal><div>Foobar</div></Modal>);
  expect(component).toMatchSnapshot();
});
