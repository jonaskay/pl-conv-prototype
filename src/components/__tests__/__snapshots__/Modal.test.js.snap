// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a modal 1`] = `
<div
  className="Modal"
>
  <div
    className="Modal-dialog"
  >
    <div>
      Foobar
    </div>
  </div>
</div>
`;
